lib/govuk_content_schemas/schema_combiner.rb

```diff
@@ -71,19 +71,12 @@ def merge_schemas(base_schema, other)
   end
 
   def add_format_field(schema)
-    properties = schema.delete('properties')
-    required = schema.delete('required') || []
-    schema['additionalProperties'] = true
-    schema['oneOf'] = [
-      {
-        "properties" => {
-          "document_type" => document_type,
-          "schema_name" => format_with_name
-        }.merge(properties),
-        "required" => ['document_type', 'schema_name'] + required,
-        "additionalProperties" => false
-      }
-    ]
+    schema["properties"]["document_type"] = document_type
+    schema["properties"]["schema_name"] = format_with_name
+    schema["required"] ||= []
+    schema["required"] << "document_type"
+    schema["required"] << "schema_name"
```

Comment:
Maybe it doesn't apply, but do we need to call `uniq` on `schema['required']` after we add both document_type and schema_name? From line 76 it looks like it could have stuff already there, is there a risk document_type and schema_name are already in that array?